src/mesh/nemesis_io_helper.C

```diff
@@ -917,7 +917,7 @@ void Nemesis_IO_Helper::write_exodus_initialization_info(const MeshBase& pmesh,
 {
   // This follows the convention of Exodus: we always write out the mesh as LIBMESH_DIM-dimensional,
   // even if it is 2D...
-  this->num_dim = pmesh.spatial_dimension();
+  this->num_dim = LIBMESH_DIM;
```

Comment:
Definitely a short-term fix to maintain the original behavior.  Longer term, the right thing to do is let the user choose which dimension to use when writing the mesh, similarly to ExodusII_IO.